stingray/lightcurve.py

```diff
@@ -101,18 +131,46 @@ def __init__(self, time, counts, input_counts=True, gti=None):
             raise StingrayError("A single or no data points can not create "
                                 "a lightcurve!")
 
+        if err is not None:
+            if not np.all(np.isfinite(err)):
+                raise ValueError("There are inf or NaN values in "
+                                 "your err array")
+        else:
+            if statistic not in valid_statistics:
+                # statistic set can be increased with other statistics
+                raise StingrayError("Statistic not recognized."
+                                    "Please select one of these: ",
+                                    "{}".format(valid_statistics))
+            if statistic == 'poisson':
```

Comment:
it might be useful to transform `statistic` to `statistic.lower()` somewhere, to make sure users don't get error messages just because they wrote `Poisson` or `poissoN` or whatever by accident.